File: FloydWarshell/FloydWarshell.go
package main

import "fmt"

var inf = 999
var v int

func floydWarshell(array [][]int) {
	var dist [100][100]int

	for i := 0; i < v; i++ {
		for j := 0; j < v; j++ {
			dist[i][j] = array[i][j]
		}
	}

	for k := 0; k < v; k++ {
		for i := 0; i < v; i++ {
			for j := 0; j < v; j++ {
				if dist[i][k]+dist[k][j] < dist[i][j] {
					dist[i][j] = dist[i][k] + dist[k][j]
				}
			}
		}
	}

	for i := 0; i < v; i++ {
		for j := 0; j < v; j++ {
			if dist[i][j] == inf {
				fmt.Printf("INF\t")
			} else {
				fmt.Printf("%d\t", dist[i][j])
			}
		}
		fmt.Printf("\n")
	}

}

func main() {

	var array [][]int
	fmt.Printf("Enter the no. of vertices: ")
	fmt.Scanf("%d", &v)
	array = make([][]int, v)
	fmt.Printf("for infinity value, use '999':\n")
	fmt.Printf("Input the graph:\n")
	for i := range array {
		fmt.Printf("Input the vertice %d:\n", i+1)
		array[i] = make([]int, v)
		for j := range array[i] {
			fmt.Scanf("%d ", &array[i][j])
		}
	}
	fmt.Printf("\n\n Shortest Path \n")
	floydWarshell(array)
}

File: Floyd-Warshell.js
function floydWarshall(graph) 
{
  let dist=[];
  let i,j,k;
  let V=4;
  let INF=99999;
	for(i=0;i<V;i++) 
		for(j=0;j<V;j++) 
			dist[i][j]=graph[i][j]; 
	for(k=0;k<V;k++) 
	{
		for(i=0;i<V;i++) 
		{ 
			for(j=0;j<V;j++) 
			{ 
				if(dist[i][k]+dist[k][j]<dist[i][j]) 
					dist[i][j]=dist[i][k]+dist[k][j]; 
			} 
		} 
	} 
	printSolution(dist); 
}

function printSolution(dist) 
{ 
	let V=4;
	  let INF=99999;
  for(let i=0;i<V;i++) 
	{ 
		for(let j=0;j<V;j++) 
		{ 
			if(dist[i][j]==INF) 
        console.log(INF)
		else
        console.log(dist[i][j])
		} 
	} 
} 
	
let graph=[
    [0, 5, INF, 10], 
		[INF, 0, 3, INF], 
		[INF, INF, 0, 1], 
		[INF, INF, INF, 0]];
	console.log(floydWarshall(graph)); 

	